shortfin/CMakeLists.txt

```diff
@@ -40,7 +40,8 @@ if(NOT WIN32)
 endif()
 
 # Pins
-set(SHORTFIN_IREE_GIT_TAG "iree-2.9.0rc20241108")
+file(STRINGS ${CMAKE_CURRENT_SOURCE_DIR}/build_tools/iree_version.txt IREE_VERSION_STRING)
+set(SHORTFIN_IREE_GIT_TAG "iree-${IREE_VERSION_STRING}")
```

Comment:
We can keep the ability to pin to an arbitrary git ref (branch name, tag, or commit hash) by switching the workflows to omit `SHORTFIN_IREE_SOURCE_DIR` and let it use `FetchContent`. I'd rather do that workflow simplification instead.

I'd rather have our most tested path be the default path with the most streamlined setup, instead of the balancing act of cloning multiple repositories and pointing them at each other explicitly.

(and folding the workflows into a single workflow that uses a `matrix` would also save on duplicated code)